### ISSUE TITLE
Result of call is ignored inspection

### ISSUE THREAD

**Issue by dlsniper** [go-lang-plugin-org/go-lang-idea-plugin#2285]
```go
package main

func demo() string {
	return "demo"
}

func main() {
	demo() // Result of call is ignored
}
```

**Comment by NathanBaulch** [comment]
This is especially useful for error return values which might signal an important failure that shouldn't be ignored.

**Comment by dlsniper** [comment]
Agreed. The question is: how should this be shown? Weak warning or warning? There are valid cases where one might call a function without needing the result of the call. Any opinions?

**Comment by stuartcarnie** [comment]
It it possible to have multiple inspections, one specifically for ignoring error values?

**Comment by sjamesr** [comment]
I think we have to be judicious with this warning otherwise it's going to
appear everywhere. For example fmt.Print* functions return (int, error) but
these are rarely checked.

**Comment by stuartcarnie** [comment]
Agree, @sjamesr – it's a double-edged sword

**Comment by dlsniper** [comment]
We can "disable" it for common known function calls but that's why I was saying. `weak warning` might be a good fit (but iirc in Java this is a ` warning `).

**Comment by ignatov** [comment]
It can be a disabled by default inspection but you can run it time to time in the batch mode.
